### PR TITLE
Feature 141909 readonly mode acceptance test

### PR DESCRIPTION
**Pull request by Nasutya**
Add acceptance test for readonly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,13 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 ### Changed
 * Blueprints:
     * Changed regular expression in `flexberry-core` for cyrillic support.
+* Changed componenet `flexberry-darepicker` classNames: add class `flexberry-datepicker`.
```
Comment (SeregaMatin) [inline CHANGELOG.md]: Typo flexberry-daREpicker.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,13 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+* Changed componenet `flexberry-simpledatetime`: remove readonlyObserver.
+* Component's template:
+  * Changed readonly attribute for `flexberry-darepicker`.
```
Comment (SeregaMatin) [inline CHANGELOG.md]: Same typo.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,13 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+  * Add readonly attribute for `flexberry-simpledatetime`.
+  * Add button's classes `flexberry-darepicker`.
```
Comment (SeregaMatin) [inline CHANGELOG.md]: Same typo.

File: app/templates/components/flexberry-lookup.hbs [hunk 4]
```diff
@@ -18,12 +18,12 @@
...
     }}
     <button
-      class="ui {{if (or readonly isBlocked) " disabled"}} {{chooseButtonClass}} {{if (or modalIsBeforeToShow modalIsStartToShow) " loading"}} button"
+      class="ui {{if (or readonly isBlocked) " disabled"}} {{chooseButtonClass}} {{if (or modalIsBeforeToShow modalIsStartToShow) " loading"}} choose-button"
       {{action "choose" chooseData}}>
       {{{chooseText}}}
     </button>
     <button
-      class="ui {{if (or readonly isBlocked) " disabled"}} {{removeButtonClass}} button"
+      class="ui {{if (or readonly isBlocked) " disabled"}} {{removeButtonClass}} remove-button"
```
Comment (SeregaMatin) [inline app/templates/components/flexberry-lookup.hbs]: CSS-classs 'button' must be restored, 'remove-button' class must be additional, but not replacing.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+  visit(path);
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $checkbox = Ember.$('.not-in-groupedit .flexberry-checkbox');
+    assert.strictEqual($checkbox.hasClass('read-only'), true, 'Checkbox is readonly');
+
+    controller.set('readonly', false);
+    Ember.run.scheduleOnce('afterRender', () => {
+      assert.strictEqual($checkbox.hasClass('read-only'), false, 'Checkbox don\'t readonly');
+    });
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: You forget to provide same checks for component inside flexberry-groupedit.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $textbox = Ember.$('.not-in-groupedit .flexberry-textbox');
+    let $textboxInput = $textbox.children('input');
+    assert.strictEqual(Ember.$.trim($textboxInput.attr('readonly')), 'readonly', 'Textbox is readonly');
+
+    controller.set('readonly', false);
+    Ember.run.scheduleOnce('afterRender', () => {
+      assert.strictEqual($(this).is('readonly'), false, 'Textbox don\'t readonly');
+    });
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: You forget to provide same checks for component inside flexberry-groupedit.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $textarea = Ember.$('.not-in-groupedit .flexberry-textarea');
+    let $textareaInput = $textarea.children('textarea');
+    assert.strictEqual(Ember.$.trim($textareaInput.attr('readonly')), 'readonly', 'Textarea is readonly');
+
+    controller.set('readonly', false);
+    Ember.run.scheduleOnce('afterRender', () => {
+      assert.strictEqual(Ember.$.trim($textareaInput.attr('readonly')), '', 'Textarea don\'t readonly');
+    });
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: You forget to provide same checks for component inside flexberry-groupedit.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+    assert.strictEqual(Ember.$.trim($datepickerInput.attr('readonly')), 'readonly', 'Time is readonly');
+
+    let $button = $datepicker.children('.calendar');
+    assert.strictEqual($button.hasClass('link'), false, 'Datepicker hasn\'t link');
+
+    controller.set('readonly', false);
+    Ember.run.scheduleOnce('afterRender', () => {
+      assert.strictEqual(Ember.$.trim($datepickerInput.attr('readonly')), '', 'Time don\'t readonly');
+      assert.strictEqual($button.hasClass('link'), true, 'Datepicker has link');
+    });
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: You forget to provide same checks for component inside flexberry-groupedit.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $simpledatetime = Ember.$('.not-in-groupedit .flexberry-simpledatetime');
+    let $simpledatetimeInput = $simpledatetime.children('input');
+    assert.strictEqual(Ember.$.trim($simpledatetimeInput.attr('readonly')), 'readonly', 'Time is readonly');
+
+    controller.set('readonly', false);
+    Ember.run.scheduleOnce('afterRender', () => {
+      assert.strictEqual(Ember.$.trim($simpledatetimeInput.attr('readonly')), '', 'Time don\'t readonly');
+    });
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: You forget to provide same checks for component inside flexberry-groupedit.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+  visit(path);
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $dropdown = Ember.$('.not-in-groupedit .flexberry-dropdown');
+    assert.strictEqual($dropdown.hasClass('disabled'), true, 'Dropdown is readonly');
+
+    controller.set('readonly', false);
+    Ember.run.scheduleOnce('afterRender', () => {
+      assert.strictEqual($dropdown.hasClass('disabled'), false, 'Enumeration don\'t readonly');
+    });
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: You forget to provide same checks for component inside flexberry-groupedit.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+      assert.strictEqual($addButton.hasClass('disabled'), false, 'Flexberry-file\'s button \'Add\' don\'t readonly');
+
+      let $removeButton = Ember.$('div.input label.flexberry-file-remove-button');
+      assert.strictEqual($removeButton.hasClass('disabled'), true, 'Flexberry-file has button \'Remove\'');
+
+      let $uploadButton = Ember.$('div.input label.flexberry-file-upload-button');
+      assert.strictEqual($uploadButton.hasClass('disabled'), true, 'Flexberry-file has button \'Upload\'');
+
+      assert.strictEqual($downloadButton.hasClass('disabled'), true, 'Flexberry-file has button \'Download\'');
+    });
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: You forget to provide same checks for component inside flexberry-groupedit.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 12]
```diff
@@ -0,0 +1,251 @@
...
+
+    let $removeButton = Ember.$('div.input button.remove-button');
+    assert.strictEqual($removeButton.hasClass('disabled'), true, 'Flexberry-lookup\'s button \'Remove\' is readonly');
+
+    controller.set('readonly', false);
+    Ember.run.scheduleOnce('afterRender', () => {
+      assert.strictEqual($(this).is('readonly'), false, 'Lookup don\'t readonly');
+      assert.strictEqual($chooseButton.hasClass('disabled'), false, 'Flexberry-lookup\'s button \'Choose\' don\'t readonly');
+      assert.strictEqual($removeButton.hasClass('disabled'), false, 'Flexberry-lookup\'s button \'Remove\' don\'t readonly');
+    });
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: You forget to provide same checks for component inside flexberry-groupedit.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $dropdownAsLookup = Ember.$('.not-in-groupedit .flexberry-lookup');
+    let $dropdown = $($dropdownAsLookup[1]).children('.flexberry-dropdown');
+    assert.strictEqual($dropdown.hasClass('disabled'), true, 'Lookup as dropdown is readonly');
+
+    controller.set('readonly', false);
+    Ember.run.scheduleOnce('afterRender', () => {
+      assert.strictEqual($dropdown.hasClass('disabled'), false, 'Lookup as dropdown don\'t readonly');
+    });
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: You forget to provide same checks for component inside flexberry-groupedit.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 14]
```diff
@@ -0,0 +1,251 @@
...
+  });
+});
+
+test('flexberry-file on readonly edit form', (assert) => {
+  assert.expect(7);
+
+  visit(path);
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $file = Ember.$('div.input input.flexberry-file-filename-input');
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: You forget about '.not-in-groupedit' class, so now you are selecting both in same time (component simply placed on  the form and those one that is placed inside groupedit).
Selectors like 'div.input input' are useless, you can select input like that: `let $fileNameInput = Ember.$('.not-in-groupedit .flexberry-file .flexberry-file-filename-input');` or like that: `let $flexberryFile = Ember.$('.not-in-groupedit .flexberry-file'); let $fileNameInput = Ember.$(' .flexberry-file-filename-input', $flexberryFile)` same selection ways are available for component's buttons and other markup.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 15]
```diff
@@ -0,0 +1,251 @@
...
+  });
+});
+
+test('flexberry-lookup on readonly edit form', (assert) => {
+  assert.expect(6);
+
+  visit(path);
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $lookup = Ember.$('div.input input.lookup-field');
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: See upper comments (about same selectors for flexberry-file).

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 16]
```diff
@@ -0,0 +1,251 @@
...
+    assert.strictEqual(Ember.$.trim($lookup.attr('readonly')), 'readonly', 'Lookup is readonly');
+
+    let $chooseButton = Ember.$('div.input button.choose-button');
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: See upper comments (about same selectors for flexberry-file).

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js [hunk 17]
```diff
@@ -0,0 +1,251 @@
...
+    assert.strictEqual($chooseButton.hasClass('disabled'), true, 'Flexberry-lookup\'s button \'Choose\' is readonly');
+
+    let $removeButton = Ember.$('div.input button.remove-button');
```
Comment (SeregaMatin) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: See upper comments (about same selectors for flexberry-file).

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js (hunk 13 above)
Comment (Nasutya) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: Fix.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js (hunk 12 above)
Comment (Nasutya) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: Fix.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js (hunk 11 above)
Comment (Nasutya) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: Fix.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js (hunk 9 above)
Comment (Nasutya) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: Groupedit hasn't this.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js (hunk 8 above)
Comment (Nasutya) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: Fix.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js (hunk 7 above)
Comment (Nasutya) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: Groupedit hasn't this.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js (hunk 6 above)
Comment (Nasutya) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: Fix.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js (hunk 5 above)
Comment (Nasutya) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: Fix.

File: app/templates/components/flexberry-lookup.hbs (hunk 4 above)
Comment (Nasutya) [inline app/templates/components/flexberry-lookup.hbs]: Fix.

File: CHANGELOG.md (hunk 3 above)
Comment (Nasutya) [inline CHANGELOG.md]: Fix.

File: CHANGELOG.md (hunk 2 above)
Comment (Nasutya) [inline CHANGELOG.md]: Fix.

File: CHANGELOG.md (hunk 1 above)
Comment (Nasutya) [inline CHANGELOG.md]: Fix.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js (hunk 14 above)
Comment (Nasutya) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: Fix.

**Comment by Nasutya** [comment]
Fix.

File: tests/acceptance/components/readonly-test/edit-form-readonly-test.js (hunk 17 above)
Comment (Nasutya) [inline tests/acceptance/components/readonly-test/edit-form-readonly-test.js]: Fix.

**Pull request merged**